### ISSUE TITLE
Run mode on client-side

### ISSUE THREAD

**Issue by alrz** [ashmind/SharpLab#341]
Run mode has some limitations (the big one is `unsafe`), is it feasible to send back compiled binaries to browser and run as WebAssembly? (this might be related to https://github.com/ashmind/SharpLab/issues/256)

**Comment by ashmind** [comment]
Yep it is related. I would be happy to do that, but is there a Roslyn WA implementation yet, or only on Mono?

**Comment by alrz** [comment]
>Roslyn WA implementation

think that would be needed for compilation, run mode alone doesn't need it

**Comment by ashmind** [comment]
@alrz How would I get a WebAssembly from the source though?

**Comment by ashmind** [comment]
Hmm, wait, I might be able to run normal .NET IL as you said. I'll investigate.

**Comment by alrz** [comment]
This was fast: https://kjpou1.github.io/WasmRoslyn/
Source: https://github.com/kjpou1/Mono.Wasm.Samples/tree/master/WasmRoslyn

**Comment by ashmind** [comment]
@alrz Thanks, that's very interesting -- I will see if I can completely switch mirrorsharp to that approach, but that would probably be a separate ticket.

**Comment by alrz** [comment]
Does this mean we don't need any roundtripping to the server (WasmRoslyn doesn't, there's no server) -- we could simply download roslyn build artifacts from azure (https://github.com/ashmind/SharpLab/issues/362), compile and run code all on client-side? I'm thinking if that's possible we may be able to accept local builds as well?

**Comment by ashmind** [comment]
I don't see why not (if Azure builds have proper CORS headers). I _think_ I still need server part for some of the Run features, but maybe not even that.

I'll deprioritize this ticket and look into mirrorsharp one instead — no reason to design a way to download an assembly if the assembly will be built on local later on.
